src/randgraphs.jl

```diff
@@ -193,6 +193,191 @@ function barabasi_albert(n::Integer, k::Integer; is_directed=false, seed::Int=-1
+    end
+    return g
+end
+
+function static_fitness_model{T<:Real,S<:Real}(m::Int, fitness_out::Vector{T}, fitness_in::Vector{S}; seed::Int=-1)
+    @assert(m >= 0, "invalid number of edges")
+    n = length(fitness_out)
+    @assert(length(fitness_in)==n, "fitness_in must have the same size as fitness_out")
+    if m == 0
+        return DiGraph(n)
+    end
+    # sanity check for the fitness
+    @assert(minimum(fitness_out) >= zero(eltype(fitness_out)) && minimum(fitness_in) >= zero(eltype(fitness_in)), "fitness scores must be non-negative")
+    # avoid getting into an infinite loop when too many edges are requested
+    outnodes = 0
+    innodes = 0
+    nodes = 0
+    for i in eachindex(fitness_out)
+        if fitness_out[i] > zero(T)
+            outnodes += 1
+        end
+        if fitness_in[i] > zero(S)
+            innodes += 1
+        end
+        if fitness_out[i] > zero(T) && fitness_in[i] > zero(S)
+            nodes += 1
+        end
+    end
+    max_no_of_edges = outnodes*innodes - nodes
+    @assert(m <= max_no_of_edges, "too many edges requested")
+    # calculate the cumulative fitness scores
+    cum_fitness_out = cumsum(fitness_out)
+    cum_fitness_in = cumsum(fitness_in)
+    max_out = cum_fitness_out[end]
+    max_in = cum_fitness_in[end]
+    edges = Set{Edge}()
+    rng = getRNG(seed)
+    while m > 0
+        source = searchsortedfirst(cum_fitness_out, rand(rng)*max_out)
+        target = searchsortedfirst(cum_fitness_in, rand(rng)*max_in)
+        # skip if loop edge
+        (source == target) && continue
+        edge = Edge(source, target)
+        # is there already an edge? If so, try again
+        in(edge, edges) && continue
+        # insert the edge
+        push!(edges, edge)
+        m -= 1
+    end
+    # create the graph
+    g = DiGraph(n)
+    for edge in edges
+        add_edge!(g, edge)
+    end
+    return g
+end
+
+"""
+    function static_scale_free(n::Int, m::Int, alpha::Float64; seed::Int=-1, finite_size_correction::Bool=false)
+
+Generates a non-growing random graph with expected power-law degree distributions. 
+`n` is the number of nodes in the generated graph. `m` is the number of edges in the generated graph. 
+`alpha` is the power law exponent of digree distribution. 
+`finite_size_correction` determines whether to use the proposed finite size correction of Cho et al. 
+Time complexity is Time complexity is O(|V| + |E| log |E|).
+
+References:
+
+* Goh K-I, Kahng B, Kim D: Universal behaviour of load distribution in scale-free networks. Phys Rev Lett 87(27):278701, 2001.
+
+* Chung F and Lu L: Connected components in a random graph with given degree sequences. Annals of Combinatorics 6, 125-145, 2002.
+
+* Cho YS, Kim JS, Park J, Kahng B, Kim D: Percolation transitions in scale-free networks under the Achlioptas process. Phys Rev Lett 103:135702, 2009.
+"""
+function static_scale_free(n::Int, m::Int, alpha::Float64; seed::Int=-1, finite_size_correction::Bool=false)
```

Comment:
Is there no way to convert this definition of `static_scale_free` to the one below so that you don't have to repeat large sections of code?